components/support/utils/src/main/java/mozilla/components/support/utils/DomainMatcher.kt

```diff
@@ -0,0 +1,49 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.support.utils
+
+import java.net.URL
+
+const val WWW_PREFIX_OFFSET = 4
+
+// FIXME implement Fennec-style segment matching logic
+// See https://github.com/mozilla-mobile/android-components/issues/1279
+object DomainMatcher {
```

Comment:
I wonder whether we actually need this Singleton object or if we could just create a function that can be imported :)